core/src/main/scala/kafka/log/LogConfig.scala

```diff
@@ -232,7 +232,7 @@ object LogConfig {
     }
   }
 
-  private val configDef: LogConfigDef = {
+  val configDef: LogConfigDef = {
```

Comment:
Can you elaborate a little more on what you need in the internal project? Generally, we avoid exposing mutable instances like this. This PR is exposing the mutable map for `serverDefaultConfigNames` and the mutable config definition.

Could we be a bit more selective in what we expose and still satisfy your requirements?